numba/tests/test_chrome_trace.py

```diff
@@ -0,0 +1,23 @@
+from .test_cli import run_cmd
+import json
+from numba.tests.support import TestCase, override_env_config
+import os
+import sys
+from tempfile import TemporaryDirectory
+
+
+class TestChromeTraceModule(TestCase):
+    """
+    Test chrome tracing generated file(s).
+    """
+
+    def test_trace_output(self):
+        with TemporaryDirectory() as tmpdir:
+            path = os.path.join(tmpdir, "test_trace.json")
+            with override_env_config('NUMBA_CHROME_TRACE', path):
+                from .chrome_trace_usecase import __file__
+                cmdline = [sys.executable, __file__]
+                run_cmd(cmdline)
```

Comment:
Can probably make use of `numba.tests.support::run_in_subprocess` and define the contents of `chrome_trace_usecase.py` as a string. Perhaps something like this produces the equivalent:
```python
    def test_trace_output(self):
        code = """
                from numba import njit
                import numpy as np

                x = np.arange(100).reshape(10, 10)

                @njit
                def go_fast(a):
                    trace = 0.0
                    for i in range(a.shape[0]):
                        trace += np.tanh(a[i, i])
                    return a + trace

                go_fast(x)
                """

        src = dedent(code)
        with TemporaryDirectory() as tmpdir:
            path = os.path.join(tmpdir, "test_trace.json")
            env = os.environ.copy()
            env['NUMBA_CHROME_TRACE'] = path
            run_in_subprocess(src, env=env)
            with open(path) as file:
                jfile = json.load(file)
                self.assertTrue(jfile)
```